### PR TITLE
Add tests for members

### PR DESCRIPTION
**Pull request by rlamy**
This adds tests for all working member kinds. Note that `OBJECT` and `OBJECT_EX` can't work yet.

The tests follow what CPython does, even though it's rather inconsistent for the exception types that are raised for invalid operations.

### CODE REVIEW

**Comment by thedrow** [comment]
> what about having `FooObject` which contains one distinct field for every type we want to check? This would be:
> 
> * more direct, because it's "straight" C code without having to instantiate it multiple times
> * faster, because you compile only one module
> * with better tracebacks, because you would see things like `assert Foo.f_unsigned_long_long == 42` instead of a generic `f.member`, without knowing which type is failing
> * easier to test, because if we explicitly write an assert for each type, we can also check corner cases such as overflows instead of using always `42`

Alternatively, you can expect a specific exception message.

File: test/test_hpytype.py [hunk 1]
```diff
@@ -221,6 +221,338 @@ def test_HPyDef_Member(self):
...
+        with pytest.raises(TypeError):
+            del foo.BOOL_member
+
+        assert foo.NONE_member is None
+        with pytest.raises((SystemError, TypeError)):  # CPython quirk/bug
+            foo.NONE_member = None
+        with pytest.raises(TypeError):
+            del foo.NONE_member
+
+    def test_HPyDef_Member_readonly(self):
```
Comment (hodgestar) [inline test/test_hpytype.py:454]: I guess this test should be named `test_HPyDef_Member_readonly_others` for consistency?

File: test/test_hpytype.py [hunk 2]
```diff
@@ -221,6 +221,338 @@ def test_HPyDef_Member(self):
...
+            {
+                FooObject *foo;
+                HPy h_obj = HPy_New(ctx, cls, &foo);
+                if (HPy_IsNull(h_obj))
+                    return HPy_NULL;
+                foo->member = 42;
+                return h_obj;
+            }
+
+            HPyDef_MEMBER(Foo_member, "member", HPyMember_%(kind)s, offsetof(FooObject, member), 1)
```
[the remark this answers is not in the document]
Comment (hodgestar) [reply test/test_hpytype.py:320]: Lol, indeed. But what does it *do*? ;)

[the remark this answers is not in the document]
Comment (antocuni) [reply test/test_hpytype.py:320]: ah, I see your point. Hint: we can rewrite it this way (and probably we should also **document** it this way):
```c
HPyDef_MEMBER(Foo_member, "member", HPyMember_%(kind)s, offsetof(FooObject, member), .readonly=1)
```

it works thanks to the magic of ``__VA_ARGS__`` here:
https://github.com/hpyproject/hpy/blob/11af3c91129226431d75b475cba858007e71969c/hpy/devel/include/common/hpydef.h#L142-L151

[the remark this answers is not in the document]
Comment (hodgestar) [reply test/test_hpytype.py:320]: Woot. The `.readonly=1` looks good and prevents accidentally passing the flag into the wrong field.

[the remark this answers is not in the document]
Comment (arigo) [reply test/test_hpytype.py:320]: Who knew, this looks like a way to write a macro `foo()` that looks like a function but can be optionally called with keyword arguments, like Python...

**Review by hodgestar — APPROVED** [review]
I implemented the changes I requested and plan to merge if build passes.

**Comment by rlamy** [comment]
Thanks @hodgestar !
